tests/MongoDB.Driver.Tests/MongoUrlBuilderTests.cs

```diff
@@ -1206,6 +1206,25 @@ public void TestWTimeout_Range()
             builder.WTimeout = TimeSpan.FromSeconds(1);
         }
 
+        [Theory]
+        [InlineData("mongodb://localhost", "mongodb://localhost")]
+        [InlineData("mongodb://localhost/?ssl=false", "mongodb://localhost")]
+        [InlineData("mongodb://localhost/?ssl=true", "mongodb://localhost/?ssl=true")]
+        [InlineData("mongodb://localhost:27018", "mongodb://localhost:27018")]
+        [InlineData("mongodb://localhost:27018/?ssl=false", "mongodb://localhost:27018")]
+        [InlineData("mongodb://localhost:27018/?ssl=true", "mongodb://localhost:27018/?ssl=true")]
+        [InlineData("mongodb+srv://localhost", "mongodb+srv://localhost")]
+        [InlineData("mongodb+srv://localhost/?ssl=false", "mongodb+srv://localhost/?ssl=false")]
+        [InlineData("mongodb+srv://localhost/?ssl=true", "mongodb+srv://localhost")]
+        public void ToString_should_return_expected_result_for_scheme_port_and_ssl(string connectionString, string expectedResult)
+        {
+            var subject = new MongoUrlBuilder(connectionString);
+
+            var result = subject.ToString();
+
+            result.Should().Be(expectedResult);
+        }
+
```

Comment:
I completely replaced the test from the pull request with this test that tests multiple scenarios.